scripts/prune_lm_dir.py

```diff
@@ -21,6 +22,17 @@
                     'prune*X, with X <= 1.0, tells it to prune with X times the threshold '
                     'specified with the --threshold option.  EM specifies one iteration of '
                     'E-M on the model. ')
+parser.add_argument("--target-num-ngrams", type=int, default=0,
+                    help="Target num-ngrams of final LM after pruning. "
+                    "If setting this to a positive value, the --steps would be "
+                    "ignored and a few steps may be worked out util the num-ngrams "
+                    "of pruned LM match the target-num-ngrams.")
+parser.add_argument("--tolerance", type=float, default=0.05,
+                    help="Tolerance of actual num-ngrams of final LM and the target-num-ngrams. "
+                    "This is only relevant if --target-num-ngrams is specified.")
+parser.add_argument("--max-iter", type=int, default=20,
+                    help="Max iterations allowed to find the threshold for target-num-ngrams LM. "
+                    "This is only relevant if --target-num-ngrams is specified.")
```

Comment:
I think you should modify how the threshold is specified.
Having it as a positional argument on the command line is no longer clear given that it has two meanings depending whether --target-num-ngrams is true.

I would have two options instead:
   --final-threshold, for use when --target-num-ngrams is not specified,
 and
   --initial-threshold, for use when --target-num-ngrams is specified.
Obviously this will involve changing all the example scripts.
You can set a reasonable default value to --initial-threshold, one that will work in most situations (e.g. 0.25).